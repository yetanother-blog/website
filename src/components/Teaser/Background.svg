<svg
    width="422"
    height="346"
    viewBox="0 0 422 346"
    xmlns="http://www.w3.org/2000/svg"
>
    <path
    d="M404 163.14C404 271.589 288.225 305 179.556 305C70.8864 305 0 278.682 0 170.233C0 82.5031 58.5419 0 167.211 0C275.88 0 404 54.6903 404 163.14Z"
    fill="currentColor"
    >
    <animate
        id="o1"
        dur="6s"
        repeatCount="indefinite"
        fill="freeze"
        attributeName="d"
        values="M404 163.14C404 271.589 288.225 305 179.556 305C70.8864 305 0 278.682 0 170.233C0 82.5031 58.5419 0 167.211 0C275.88 0 404 54.6903 404 163.14Z;M422 176.894C422 285.343 320.669 346 212 346C103.331 346 0.5 307.843 0.5 199.394C0.5 111.664 140.331 0.893555 249 0.893555C357.669 0.893555 422 68.4443 422 176.894Z;M404 163.14C404 271.589 288.225 305 179.556 305C70.8864 305 0 278.682 0 170.233C0 82.5031 58.5419 0 167.211 0C275.88 0 404 54.6903 404 163.14Z"
    />
    </path>
</svg>