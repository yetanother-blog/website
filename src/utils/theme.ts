import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  fontSizes: ['12px', '14px', '16px', '20px', '24px', '32px'],
  colors: {
    primary: '#00FF00',
    secondary: '#FB00FF',
    grey500: '#1C1C1C',
    grey400: '#ACADAC',
    grey300: '#F6F6F6',
  },
  breakpoints: ['576px', '768px', '992px', '1200px'],
  space: {
    zero: 0,
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 20,
    xl: 40,
    xxl: 60,
    xxxl: 120,
  },
  radii: {
    s: 3,
  },
  width: {
    m: '620px',
    l: '940px',
    xl: '1200px',
  },
};
