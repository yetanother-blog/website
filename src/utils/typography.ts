import { CSSObject } from "styled-components";
import Typography from "typography";

const typography = new Typography({
  googleFonts: [
    {
      name: "Source Sans Pro",
      styles: ["400", "600"]
    },
    {
      name: "Source Code Pro",
      styles: ["600", "700"]
    }
  ],
  baseFontSize: "20px",
  baseLineHeight: 1.85,
  bodyFontFamily: ["Source Sans Pro", "sans-serif"],
  boldWeight: "600",
  overrideStyles: ({ adjustFontSizeTo }, options, styles) => ({
    a: {
      fontFamily: `"Source Code Pro", "sans-serif"`
    }
  })
});

// Hot reload typography in development.
if (process.env.NODE_ENV !== `production`) {
  typography.injectStyles();
}

export default typography;

export const rhythm = typography.rhythm;
export const scale = typography.scale;

type StyledScale = (values: number) => CSSObject;
export const styledScale = scale as StyledScale;
