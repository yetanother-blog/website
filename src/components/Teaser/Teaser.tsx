import React from 'react';
import Background from './Background.svg';
import styled, { keyframes, useTheme } from 'styled-components';
import { Box } from '../Box/Box';
import Browser from './Browser.svg';
import { Typography } from '../Typography/Typography';

const StyledBackground = styled(Background)`
  color: ${(props) => props.theme.colors.grey300};
`;

const rotate = keyframes`
  0%, 100% {
    transform: translate(40px, -50px);
  }
  50% {
    transform: translate(44px, -42px);
  }
`;

const StyledWrapper = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  margin-bottom: ${(props) => props.theme.space.l}px;
`;

const StyledBrowser = styled(Browser)`
  position: absolute;
  transform: translate(40px, -50px);
  animation: ${rotate} 10s linear infinite;
`;

const StyledTypographyWrapper = styled.div`
  position: relative;
  display: block;
  background-color: white;
  padding: 10px;

  &:after {
    content: '';
    background-color: ${(props) => props.theme.colors.secondary};
    position: absolute;
    bottom: -6px;
    left: -6px;
    width: 100%;
    height: calc(100% - 50px);
    z-index: -1;
  }
`;

export const Teaser: React.FC = ({ children }) => {
  const theme = useTheme();
  return (
    <Box
      display="flex"
      justifyContent="center"
      flexDirection="column"
      mb={theme.space.xl}
    >
      <StyledWrapper>
        <StyledBrowser />
        <StyledBackground />
      </StyledWrapper>
      <StyledTypographyWrapper>
        <Typography
          variant="title"
          as="h2"
          fontWeight={600}
          fontFamily="Source Code Pro"
          marginBottom="0"
          textAlign="center"
        >
          {children}
        </Typography>
      </StyledTypographyWrapper>
    </Box>
  );
};
