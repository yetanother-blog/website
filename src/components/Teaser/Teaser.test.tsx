import React from 'react';
import { render } from '@testing-library/react';
import { Teaser } from './Teaser';
import { ThemeContext } from 'styled-components';
import { theme } from '../../utils/theme';

describe('Teaser component', () => {
  it('should render a teaser w/ content', () => {
    const { container } = render(
      <ThemeContext.Provider value={theme}>
        <Teaser>Hello World</Teaser>
      </ThemeContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
