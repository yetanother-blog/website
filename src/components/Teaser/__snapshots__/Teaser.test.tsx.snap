// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teaser component should render a teaser w/ content 1`] = `
<div>
  <div
    class="sc-bdvvaa jpUCIR"
    display="flex"
  >
    <span
      class="sc-iCfLBT iWpJp"
    >
      <svg
        class="sc-pVTma ghdccg"
        data-file-name="SvgBrowser"
      />
      <svg
        class="sc-furvIG ezfHtA"
        data-file-name="SvgBackground"
      />
    </span>
    <div
      class="sc-jrQzUz bgLYmB"
    >
      <h2
        class="sc-gsDJrp ebQNtU"
        font-family="Source Code Pro"
        font-weight="600"
      >
        Hello World
      </h2>
    </div>
  </div>
</div>
`;
